app/services/ingest_service.rb

```diff
@@ -0,0 +1,68 @@
+class IngestService
+  def initialize(logger = nil)
+    @logger = logger || Logger.new(STDOUT)
+  end
+
+  def minimal_record(klass, user, attributes)
+    default_attributes = { rights_statement: 'http://rightsstatements.org/vocab/NKC/1.0/',
+                           visibility: Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PUBLIC }
+    r = klass.new
+    r.attributes = default_attributes.merge(attributes)
+    r.apply_depositor_metadata user
+    r.apply_remote_metadata if r.source_metadata_identifier
+    r.save!
+    Workflow::InitializeState.call(r, workflow_name, 'final_review')
+    @logger.info "Created #{klass}: #{r.id} #{attributes}"
+
+    r
+  end
+
+  def ingest_dir(dir, bib, user)
+    klass = choose_class(Dir["#{dir}/*"].first)
+    attribs = bib.nil? ? { title: [File.basename(dir)] } : { source_metadata_identifier: bib }
+    r = minimal_record klass, user, attribs
+    members = []
+    Dir["#{dir}/*"].sort.each do |f|
+      if File.directory? f
+        members << ingest_dir(f, nil, user)
+      else
+        members << ingest_file(r, f, user, file_set_attributes, title: [File.basename(f)])
```

Comment:
Looks like the `file_set_attributes` are being passed as file_options (visibility, lease, embargo) instead of attributes. Would something like this work? 
```
members << ingest_file(r, f, user, {}, file_set_attributes.merge({ title: [File.basename(f)] }))
```